dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -353,6 +354,11 @@ private boolean shouldAttemptToRetry(Throwable throwable) {
                             // String matches CJR RemotingOkHttpCall.shouldRetry
                             && socketTimeout.getMessage().contains("connect timed out");
                 }
+                if (throwable instanceof NoHttpResponseException) {
```

Comment:
might be leaking implementation details (apache client) here, open to suggestions.